tests/func/test_stage.py

```diff
@@ -178,3 +181,29 @@ def test_meta_is_preserved(tmp_dir, dvc):
 
     new_data = load_stage_file(stage.path)
     assert new_data["meta"] == data["meta"]
+
+
+def test_warn_on_no_out_checksum(tmp_dir, dvc, tmp_path_factory, caplog):
+    tmp_dir.dvc_gen("file", "file content")
+
+    local_remote = tmp_path_factory.mktemp("local_remote")
+    with dvc.config.edit() as conf:
+        from dvc.utils import fspath
+
+        conf["remote"]["local_remote"] = {"url": fspath(local_remote)}
+        conf["core"]["remote"] = "local_remote"
+
+    stage_content = load_stage_file("file.dvc")
+    first(stage_content["outs"])["md5"] = None
+    dump_stage_file("file.dvc", stage_content)
+
+    with caplog.at_level(logging.WARNING, logger="dvc"):
+        dvc.push()
+
+    assert first(caplog.messages) == (
+        "Output 'file'(Stage: 'file.dvc') is missing version info. "
+        "Cache for it will not be collected. "
+        "Use `dvc repro` to get your pipeline up to date.\n"
+        "You can also use `dvc commit file.dvc` to associate existing 'file' "
+        "with `file.dvc`."
+    )
```

Comment:
Maybe let's narrow it down to get_used_cache unit test for output?

Reply:
makes sense